crates/astria-sequencer-relayer/src/types.rs

```diff
@@ -75,6 +75,13 @@ impl Namespace {
     }
 }
 
+impl fmt::Debug for Namespace {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // FIXME: `hex::encode` does an extra allocation which could be removed
+        f.write_str(&hex::encode(self.0))
```

Comment:
This is breaking with rust convention and should not be done.

Also, this is the same as the `Display for Namespace` impl. Why not use that?

Reply:
the hashmap requires debug, which causes the namespaces to be printed not as hex strings

Reply:
I understand - this is fundamentally about tracing not being able to easily generate fields for  more complex structures. This should all be fixed with the valuable trait/crate. But it's not really moving anywhere.

For the time being you can work around it with either what I have done in https://github.com/astriaorg/astria/tree/superfluffy/submissions_logging (providing a wrapper to skip building the hashmap), or/in addition creating something like a `struct DisplayJson<a, T>(T)` and `impl<'a, T> Display for DisplayJson<'a, T> where T: Serialize` or something like that.

We should probably do something like the latter at some point because this issue will start popping up everywhere in the codebase at some point.

https://github.com/tokio-rs/tracing/discussions/1906